cpp/include/raft/neighbors/detail/ivf_pq_search.cuh

```diff
@@ -704,11 +708,11 @@ inline auto get_max_batch_size(uint32_t k,
   auto ws_size = [k, n_probes, max_samples](uint32_t bs) -> uint64_t {
     return uint64_t(is_local_topk_feasible(k, n_probes, bs) ? k * n_probes : max_samples) * bs;
   };
-  constexpr uint64_t kMaxWsSize = 1024 * 1024 * 1024;
-  if (ws_size(max_batch_size) > kMaxWsSize) {
+  auto max_ws_size = resource::get_workspace_total_bytes(res);
```

Comment:
Why not `get_workspace_free_bytes`?

Reply:
Honestly I'm not sure which one is better here. On the one hand, `get_workspace_total_bytes` gives slightly more consistent experience (the batch sizes and thus performance don't differ depending on what code is running alongside the ivf-pq search and consuming the workspace memory); on the other hand, there could be a possible scenario where the search would throw exception due to trying to get more space than available.

Thus, I think both solutions are not perfect. Let me know if you prefer the `get_workspace_free_bytes` slightly more, I'm happy to replace it.